File: src/main/java/es/uji/sape/model/Tutor.java
package es.uji.sape.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import static es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING;

@Data
@NoArgsConstructor
public final class Tutor {

    private @NotNull String code = "";
    private @NotNull String name = "";
    private @NotNull String surname = "";
    private @NotNull Itinerary itinerary = SOFTWARE_ENGINEERING;
    private @NotNull String department = "";
    private @NotNull String office = "";
}

File: src/main/kotlin/es/uji/sape/model/Tutor.kt
package es.uji.sape.model

import es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING

data class Tutor(
    val code: String = "",
    val name: String = "",
    val surname: String = "",
    val itinerary: Itinerary = SOFTWARE_ENGINEERING,
    val department: String = "",
    val office: String = ""
)

